### DISCUSSION TITLE
need noise :)

### DISCUSSION THREAD

**Discussion by steeley** [Quefumas/gensound#20] [Ideas]
this is quite a nice library, please keep developing!
(I tried PYO which is very good but could not get it to work with pulseaudio)

White noise and pink noise are important for me, I really need these in gensound

thanks

**Comment by Quefumas** [comment]
Thanks! It's great to hear people are using this.

Actually, there already exists white noise, use `from signals import WhiteNoise`. The constructor receives a `duration` argument, which can (as usual) be float for milliseconds (for example `3e3` as scientific notation for '3 seconds'), or integer for samples. I've updated the signals page in the wiki to reflect this.

Note that this class generates the samples randomly and independently in the range `[-1,1]`, and I'm not yet 100% sure this is the best way to generate white noise in particular.

As for pink noise, I think I can implement it quite easily and push without even having to wait for the next version. In case you happen to have a handy formula for pink noise, feel free to share, that will probably speed me up considerably.

**Comment by Quefumas** [comment]
@steeley I've just committed a provisional `PinkNoise` class (you can see it next to `WhiteNoise` in `signals.py`). It will not be included in the pip installation until the next version comes out, which could take a while, but it's possible to just copy & paste into your code from `signals.py` (provided you `import numpy as np` and also `from gensound import Signal`).

Here is a usage example:

```python
from gensound import WhiteNoise, PinkNoise

# play 3 secs of pink noise followed by 3 secs of white noise
(PinkNoise(3e3) | WhiteNoise(3e3)).play()
```

**Comment by steeley** [comment]
thanks looks good
